### DISCUSSION TITLE
[Solved]Getting Sleep (And hopefully hibernation) to work

### DISCUSSION THREAD

**Discussion by misterblack0101** [AndresGarciaSobrado91/MSI-Modern15-Hackintosh#18] [Q&A] [answered]
Hey, I'm unable to get my laptop to sleep, or hibernate.
I read that hibernation is buggy, and might not work, and I can let that go.
But when I try to make my laptop sleep, it doesn't. with the lid down, it should've gone to sleep, but it didn't and used 15% battery in just 1.5hrs!
I tried all the methods on dortania's guide, and still can't figure out what's going/ how to fix it.
![Screenshot 2021-10-30 at 6 54 29 PM](https://user-images.githubusercontent.com/71558331/139534463-2eb070e4-3d0d-46f0-b374-2e696de3c159.png)
Also, normally during sleep, the indicator on my laptop's side just blinks. Right now, The laptop light starts to blink after 15 secs of putting the lid down, and the immediately after the first blink, the keyboard lights turn back on, and the blinking ld is constant again. (the keyboard lights were off during the 15 sec period).
Edit: Sleep from apple logo  menu gives the same result. (i.e. not working)

**Comment by AndresGarciaSobrado91** [comment]
Is your laptop exactly like mine? Same specs?

**Reply by misterblack0101** [reply]
The ports and screen are different.
The ports were fixed by the usbmap, and  the screen is 14", instead of 15.  Here's [my model](
https://dl.flipkart.com/s/Jof560uuuN)

**Comment by misterblack0101** [comment, marked as answer]
Update: Sleep works with usbinjectall kext. Tried USB mapping twice, but those mapped kexts again broke the sleep. All the ports are working with usbinjectall as well as my usb map. will try creating custom ssdts for usbinjectall. 
Update 2: Successfully mapped usb using hackintool. Now, all the ports, as well as the sleep works.
@AndresGarciaSobrado91 could you please close this thread? Thanks😁

**Reply by AndresGarciaSobrado91** [reply]
Excellent! 👍  Enjoy your Hackintosh! Regards.
